### ISSUE TITLE
Implement IE11 notification solution/banner with final placement and copy

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#9075]
## Description
Follow on to discovery story #8852 

### Overview
- Conditionally create, append, and load the generated web-components code, as well as the web component that will display the IE11 (or below) user warning notification only when the user is on an IE11 or older IE browser.

### Open Questions
- Where in the header should the Warning Alert appear?
  - A: Collab Cycle confirmed: use the Full Width Alert as Aslan mocked up.
- What is the copy which should be displayed within the [Warning Alert](https://design.va.gov/storybook/?path=/docs/components-va-alert--fullwidth)?
  - A: Sitewide Content Team has an issue to create final copy. To avoid delay/blocker, we will create a new issue to update our copy when they provide that. For now, use the placeholder copy that Aslan provided.
    - headline: "Internet Explorer 11 will soon be unsupported"
    - message: "You will need to switch to use Chrome, Firefox, or Safari."
- Does this require a design review?
  - A: No additional design review needed.

### Implementation
- **Legacy pages** - the solution is the same for both mobile/desktop layouts & implemented only once.
  - vets-website contains a file, src/applications/proxy-rewrite/proxy-rewrite-entry.jsx, which contains logic that will run on any legacy page - this is where we’ll add logic which appends & loads the available generated web-components JS & CSS files which are built and stored in S3, as well as the [Warning Alert Web Component](https://design.va.gov/storybook/?path=/docs/components-va-alert--fullwidth) from the shared component library.
    - Create a function(s), as needed, for the asset & element creation & append logic.
      - For the web components
        - create a `<script />` tag w/ a src of /generated/web-components.entry.js
        - create a `<link />` tag w/ an href of /generated/web-components.css
        - and then append to the `<head />` of the HTML document.
      - For the Warning Alert web component
        - create a new set of elements that will implement the following code found on the web components [Storybook page](https://design.va.gov/storybook/?path=/docs/components-va-alert--warning). See the code snippet below.
        - and then append it to the desired (TBD) location within the page header markup.
      - There’s a function within the file, proxy-rewrite-entry.jsx, called main() - this is where we’ll conditionally invoke the logic just created which is 100% dependent on the user’s browser - IE11 (or older)
        - The logic for the browser detection shall be determined by the developer working this ticket.
- **Modern pages** - The shared web components are already available to all modern pages.
  - tablet/desktop layout
     - content-build contains a file, src/site/includes/top-nav.html, which is loaded when the user is on a tablet/desktop layout - this is where we’ll add the Warning Alert Web Component (see the code snippet below.)
     - The Warning Alert should be conditionally added whenever IE11 (or older) browser detection occurs.
  - mobile layout
     - While a user reaching a mobile layout on IE11 (or older) has been deemed an edge case, the recommendation is to implement this for consistency, but more importantly for future maintenance purposes - any developer may wonder why mobile was left out in the future. Implementation will avoid confusion down the road, and this is an easy lift.
     - content-build contains a file, src/platform/site-wide/header/components/Header/index.js, which is loaded when the user is on a mobile layout (the screen aspect ration is determined to be a mobile layout) - this is where we'll conditionally add the [Warning Alert Web Component](https://design.va.gov/storybook/?path=/docs/components-va-alert--fullwidth) - same conditions as stated above.

```
// Warning Alert
<va-alert
  close-btn-aria-label="Close notification"
  status="warning"
  visible
>
  <h3 slot="headline">
    Alert headline
  </h3>
  <div>
    This is an alert
  </div>
</va-alert>
```

## Acceptance Criteria
- [ ] All legacy pages should display the Warning Alert in the header whenever a user is browsing from any Internet Explorer browser (IE11 or below) on a tablet or desktop device.
- [ ] All modern pages should display the Warning Alert in the header whenever a user is browsing from any Internet Explorer browser (IE11 or below) on a tablet or desktop device.
- [ ] Requires design review?

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by wesrowe** [comment]
@wesrowe to engage:
- [ ] Collab Cycle for design
- [ ] Copy from SW Content – open issue with them

**Comment by jilladams** [comment]
Per sprint planning: 
* Collab cycle required
* Copy needs to come from Sitewide Content team
* Technical bits can get started, but final implementation is blocked by collab cycle final answers and content, so adding dependency on #9210

**Comment by jcklpe** [comment]
Hi-fi Wireframe:
![image](https://user-images.githubusercontent.com/9451726/171478759-21855ef3-cab4-4171-9b08-740629d1293c.png)


Wireframes for design are from design intent ticket here: https://github.com/department-of-veterans-affairs/va.gov-team/issues/41913

**Comment by kind-claudio** [comment]
Status update - remaining work:
- Need to test one application view (> tablet/modern page) which only runs in a prod environment - figuring this one out.
- 3 eslint errors remaining to fix - looking up docs for eslint on these ones.
- update 1 unit test for the `Header.jsx` mobile-only widget.

**Comment by jilladams** [comment]
Per Dave:  Because header is injected, it will appear in context we cannot control. If we release, and it misbehaves in ways we do not expect, we need to be ready for that. If no feature flag, we will have to roll back, which is ok, we just need to be prepared and coordinate merge / deploy to account for that risk. Code will be ready today potentially; should not merge for Friday deploy. Can merge for Mon 6/13 deploy.

## Staging pre-release testing
* In IE11, alert appears - Dave can help verify, or IE is available via CAG
* In non-IE11 browsers, alert does not appear

## Prod testing 
After daily deploy, check header / footer on VA.gov.
EOD PT / next morning: check TeamSite propagation for injected header/footer)
* In IE11, alert appears - Dave can help verify, or IE is available via CAG
* In non-IE11 browsers, alert does not appear
* Include OPIA validation
* VA Health website - 

After launch, Dave will notify Web Governance Board that we've released.

**Comment by jilladams** [comment]
## Final copy 
From https://github.com/department-of-veterans-affairs/va.gov-team/issues/41914

So here's the whole text, with this revision:
You'll need to use a different web browser soon
You're using Internet Explorer 11 right now to access [VA.gov](http://va.gov/). Microsoft will stop supporting that browser on June 15, 2022. This means that you'll need to switch to another browser soon, like Microsoft Edge, Google Chrome, Mozilla Firefox, or Apple Safari.

After June 15, change to:
You'll need to use a different web browser
You're using Internet Explorer 11 right now to access [VA.gov](http://va.gov/). Microsoft stopped supporting that browser on June 15, 2022. This means that you'll need to switch to another browser, like Microsoft Edge, Google Chrome, Mozilla Firefox, or Apple Safari.

@wesrowe we didn't address the 6/15 cutoff. Do you want to pitch that at Dave or just cut a 2nd ticket for that update?

**Comment by wesrowe** [comment]
@jilladams and @kind-claudio, we need to do a staging review step with collab cycle, so it will be june 15 before we get this out the door.

**Comment by kind-claudio** [comment]
VA.gov

**Comment by wesrowe** [comment]
To bring it all together, final copy will be:
You'll need to use a different web browser
You're using Internet Explorer right now to access VA.gov. Microsoft stopped supporting all versions of this browser on June 15, 2022. This means that you'll need to switch to another browser, like Microsoft Edge, Google Chrome, Mozilla Firefox, or Apple Safari.

Note that VA.gov is not hyperlinked.
